tools-local/Microsoft.DotNet.Build.Tasks.Local/UploadClient.cs

```diff
@@ -142,6 +141,14 @@ public string EncodeBlockIds(int numberOfBlocks, int lengthOfId)
                     var req = new HttpRequestMessage(HttpMethod.Put, blockListUploadUrl);
                     req.Headers.Add(AzureHelper.DateHeaderString, dt1.ToString("R", CultureInfo.InvariantCulture));
                     req.Headers.Add(AzureHelper.VersionHeaderString, AzureHelper.StorageApiVersion);
+                    if(!string.IsNullOrEmpty(DetermineContentTypeBasedOnFileExtension(filePath)))
```

Comment:
I went this direction because it's more flexible, and I moved the values into methods so that they were more re-usable.